dev/TEAL.md

```diff
@@ -176,11 +222,15 @@ Some of these have immediate data in the byte or bytes after the opcode.
 | `gtxn t f` | push field F of the Tth transaction in the current group |
 | `txna f i` | push Ith value of the array field F of the current transaction |
 | `gtxna t f i` | push Ith value of the array field F from the Tth transaction in the current group |
-| `gtxns f` | push field F of the Ath transaction in the current group |
-| `gtxnsa f i` | push Ith value of the array field F from the Ath transaction in the current group |
+| `gtxns f` | push field F of the Xth transaction in the current group |
+| `gtxnsa f i` | push Ith value of the array field F from the Xth transaction in the current group |
 | `global f` | push value from globals to stack |
 | `load i` | copy a value from scratch space to the stack |
 | `store i` | pop a value from the stack and store to scratch space |
+| `gload t i` | push Ith scratch space index of the Tth transaction in the current group |
```

Comment:
These instructions are useful, but to be completely clear, it is very important that the scratch space also be accessible after the txn group is completed when we look up information about it in the indexer. This is so that we can have the smart contract have "output" in the sense of return arguments from method calls (including pure ones) and so that we can have "event logs" (a la Ethereum) by defining an ABI for the format of the final scratch space

Reply:
That is probably not the plan, as it is potentially quite large (256*4k).  We intend to have a `announce` operator or similar, but it will be more intentional, so that we aren't committing to have the scratch space kept around for a contract that might not even care.